mlflow/models/model.py

```diff
@@ -634,14 +612,14 @@ def log(
             if run_id is None:
                 run_id = mlflow.tracking.fluent._get_or_start_run().info.run_id
             mlflow_model = cls(artifact_path=artifact_path, run_id=run_id, metadata=metadata)
+            flavor.save_model(path=local_path, mlflow_model=mlflow_model, **kwargs)
             tracking_uri = _resolve_tracking_uri()
-            if (
-                (tracking_uri == "databricks" or get_uri_scheme(tracking_uri) == "databricks")
-                and kwargs.get("signature") is None
-                and kwargs.get("input_example") is None
+            # We check signature presence here as some flavors have a default signature as a
+            # fallback when not provided by user, which is set during flavor's save_model() call.
+            if getattr(mlflow_model, "signature") is None and (
```

Comment:
Do we need to use `getattr`? It looks like the `signature` attribute is always set in `__init__`.

Reply:
You're right, now it's property not key so we don't have to check